### PR TITLE
Show reaction names in diagrams

### PR DESCRIPTION
**Pull request by cmnrd**
This PR adds the "Reaction Names" diagram synthesis option, which shows the name of a reaction instead of its index. If a reaction does not have a name in LF code, its internal name is shown (see code and figure below).
```
main reactor {
  reaction hello(startup)
  reaction(shutdown)
}
```
HelloBodylessWorld

In order to align the index number that we show in the diagrams with the number in the name for anonymous reactions, I also modified the `getName()` method for reaction instances to increment the index by one. This partly addresses https://github.com/lf-lang/lingua-franca/issues/1782, but does not fully resolve the issue as there are many instances in the C code generator, where it does not use the name of the reactor instance, but instead constructs its own name.

Fixes https://github.com/lf-lang/lingua-franca/issues/1856

### CODE REVIEW

File: core/src/main/java/org/lflang/generator/ReactionInstance.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Set;
+import org.eclipse.xtext.xbase.lib.StringExtensions;
```
Comment (edwardalee) [inline core/src/main/java/org/lflang/generator/ReactionInstance.java:32]: Is this import really necessary?  It looks like it is used to just check for a null or empty string. I think that this is a rather heavyweight dependency for such small functionality.

**Review by edwardalee — CHANGES_REQUESTED** [review]
This is a very nice improvement!  One minor request inserted...

File: core/src/main/java/org/lflang/generator/ReactionInstance.java (hunk 1 above)
Comment (lhstrh) [reply core/src/main/java/org/lflang/generator/ReactionInstance.java:32]: We already have this dependency throughout the code base. If we want to remove it, we have quite a bit more work to do...

Comment (cmnrd) [reply core/src/main/java/org/lflang/generator/ReactionInstance.java:32]: Xtext is one of our core dependencies. Using its functionality throughout shouldn't do any additional harm.

**Comment by cmnrd** [comment]
@edwardalee could you approve this PR?

**Comment by lhstrh** [comment]
It would probably look better (in the diagram) without the underscore.

**Comment by cmnrd** [comment]
> It would probably look better (in the diagram) without the underscore.

You mean `reaction2` instead of `reaction_2`? That would be an easy change and probably also looks better in the generated C code.

**Comment by cmnrd** [comment]
What d you think about the bold typeface? I kept this from the index numbers, but I am wondering if it might get too much for longer names.

**Comment by lhstrh** [comment]
> What d you think about the bold typeface? I kept this from the index numbers, but I am wondering if it might get too much for longer names.

I meant with a space, but if it has to literally match the identifier then I guess the underscore is fine...

I also prefer normal over bold.

**Pull request merged**